assume/scenario/loader_pypsa.py

```diff
@@ -0,0 +1,232 @@
+# SPDX-FileCopyrightText: ASSUME Developers
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+import shutil
+from datetime import datetime, timedelta
+from pathlib import Path
+
+import pandas as pd
+import pypsa
+from dateutil import rrule as rr
+
+from assume import World
+from assume.common.forecasts import NaiveForecast
+from assume.common.market_objects import MarketConfig, MarketProduct
+
+
+async def load_pypsa_async(
+    world: World,
+    scenario: str,
+    study_case: str,
+    network: pypsa.Network,
+    marketdesign: list[MarketConfig],
+):
+    """
+    This initializes a scenario from the given pypsa grid.
+
+    Args:
+        world (World): the world to add this scenario to
+        scenario (str): scenario name
+        study_case (str): study case name
+        network (str): database uri to connect to the OEDS
+        marketdesign (list[MarketConfig]): description of the market design which will be used with the scenario
+    """
+    index = network.snapshots
+    index.freq = "h"
+    start = index[0]
+    end = index[-1]
+    sim_id = f"{scenario}_{study_case}"
+    print(f"loading scenario {sim_id}")
+
+    await world.setup(
+        start=start,
+        end=end,
+        save_frequency_hours=4,
+        simulation_id=sim_id,
+        index=index,
+    )
+    # setup eom market
+
+    mo_id = "market_operator"
+    world.add_market_operator(id=mo_id)
+    grid_data = {
+        "buses": network.buses,
+        "lines": network.lines,
+        "generators": network.generators,
+        "loads": network.loads,
+    }
+    for market_config in marketdesign:
+        market_config.param_dict["grid_data"] = grid_data
+        world.add_market(mo_id, market_config)
+
+    # naive_eom
+    default_strategy = {mc.market_id: "naive_redispatch" for mc in marketdesign}
+
+    bidding_strategies = {
```

Comment:
I am against this approach. we should allow loading from folder, where a network is placed, but also other files can be placed, such as power plants or demand units, and their bidding strategies and other things specified. The current implementation is rather a prototype one